src/test/java/picard/nio/PicardHtsPathUnitTest.java

```diff
@@ -0,0 +1,31 @@
+package picard.nio;
+
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+
+public class PicardHtsPathUnitTest {
+    @DataProvider(name="ReplaceExtensionTestInput")
+    public Object[][] getReplaceExtensionTestInput() {
+        return new Object[][] {
+                {"gs://hellbender/test/resources/hg19mini.fasta", ".fai", false, "gs://hellbender/test/resources/hg19mini.fai"},
+                {"gs://hellbender/test/resources/hg19mini.fasta", ".fai", true, "gs://hellbender/test/resources/hg19mini.fasta.fai"},
+                {"testdata/picard/sam/test.bam", ".bai", false, new File("testdata/picard/sam/test.bai").getAbsolutePath()}, // a relative input will be converted to absolute input
+                {"testdata/picard/sam/test.bam", ".bai", true, new File("testdata/picard/sam/test.bam.bai").getAbsolutePath()},
+                {"/Users/jdoe/workspace/picard/testdata/picard/sam/test.bam", ".bai", false, "/Users/jdoe/workspace/picard/testdata/picard/sam/test.bai"},
+                {"/Users/jdoe/workspace/picard/testdata/picard/sam/test.bam", ".md5", true, "/Users/jdoe/workspace/picard/testdata/picard/sam/test.bam.md5"}
+        };
```

Comment:
Suggest some test cases with query params - see my comment on the PicardHtsPath changes.